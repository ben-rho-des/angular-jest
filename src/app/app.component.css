html,
body {
    height: 100%;
}

body {
    font-family: sans-serif;
    background-image: url(/assets/where.gif);
    background-size: 100px;
    margin: 0;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
}

.containter {
    width: 600px;
    background: #eee;
    border: 1px solid #d2d2d2;
    border-radius:6px;
    padding: 2rem;
    flex: 1;
}

button {
    background: #00cc00;
    border-radius: 3px;
    border: 0;
    cursor: pointer;
    font-weight: bold;
    font-size: 100%;
    padding: 15px 30px;
}

button:hover,
button:active {
    background: #009900;
}

h1 {
    margin-top: 0;
}
