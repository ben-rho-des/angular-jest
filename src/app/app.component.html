<div class="containter">
  <h1>silly app</h1>
  <div *ngIf="step === 1;else steptwocontent">
    <app-stepone (valid)="onValid($event)"></app-stepone>
  </div>
</div>

<ng-template #steptwocontent>
  <app-steptwo></app-steptwo>
</ng-template>