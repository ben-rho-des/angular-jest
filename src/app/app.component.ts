import { Component, ViewEncapsulation } from '@angular/core';

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.css'],
  encapsulation: ViewEncapsulation.None
})
export class AppComponent {

  private step = 1;

  onValid(valid: boolean) {
    this.step = (!valid) ? 1 : 2;
  }
}
