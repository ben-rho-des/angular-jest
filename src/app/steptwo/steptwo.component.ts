import { Component, OnInit } from '@angular/core';

@Component({
  selector: 'app-steptwo',
  templateUrl: './steptwo.component.html',
  styleUrls: ['./steptwo.component.css']
})
export class SteptwoComponent implements OnInit {

  constructor() { }

  ngOnInit() {
  }

}
