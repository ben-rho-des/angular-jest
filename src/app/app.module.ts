import { BrowserModule } from '@angular/platform-browser';
import { NgModule } from '@angular/core';
import { FormsModule } from '@angular/forms';

import { AppComponent } from './app.component';
import { SteponeComponent } from './stepone/stepone.component';
import { SteptwoComponent } from './steptwo/steptwo.component';


@NgModule({
  declarations: [
    AppComponent,
    SteponeComponent,
    SteptwoComponent
  ],
  imports: [
    BrowserModule,
    FormsModule
  ],
  providers: [],
  bootstrap: [AppComponent]
})
export class AppModule { }
