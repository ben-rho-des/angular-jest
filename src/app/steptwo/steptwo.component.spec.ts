import { async, ComponentFixture, TestBed, getTestBed } from '@angular/core/testing';
import { ChangeDetectionStrategy, Component, Injector, DebugElement } from '@angular/core';
import { AppModule } from '../app.module';
import { SteptwoComponent } from './steptwo.component';

describe('SteponeComponent', () => {
    let component: SteptwoComponent;
    let fixture: ComponentFixture<SteptwoComponent>;

    setupTestBed({
        imports: [AppModule]
    });

    beforeEach(() => {
        fixture = TestBed.createComponent(SteptwoComponent);
        component = fixture.componentInstance;
    });

    it('should create', () => {
        expect(component).toBeTruthy();
    });

    it('should have the correct content', () => {
        expect(fixture.nativeElement.querySelector('p').textContent).toBe('Yay done!');
    });

    it('snapshot should pass', () => {
        expect(fixture).toMatchSnapshot();
    });
});
