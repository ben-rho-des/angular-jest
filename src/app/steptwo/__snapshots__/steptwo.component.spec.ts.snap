// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SteponeComponent snapshot should pass 1`] = `
<app-steptwo>
  <p>
    Yay done!
  </p>
</app-steptwo>
`;
